pkg/controllers/federatedcluster/clusterstatus.go

```diff
@@ -144,6 +144,10 @@ func (c *FederatedClusterController) collectIndividualClusterStatus(
 		return 0, fmt.Errorf("failed to update cluster status: %w", err)
 	}
 
+	if readyStatus != corev1.ConditionTrue {
+		c.eventRecorder.Eventf(cluster, ClusterHealthzNotOKReason, ClusterHealthzNotOKMessage, "Cluster is not ready")
```

Comment:
We should only log when there's a state transition. Otherwise an unready cluster will keep emitting events.

Reply:
done